esp-wifi/src/ble/controller/mod.rs

```diff
@@ -3,22 +3,26 @@ use embedded_io::{Error, ErrorType, Read, Write};
 use super::{read_hci, read_next, send_hci};
 use crate::{
     hal::peripheral::{Peripheral, PeripheralRef},
-    EspWifiInitialization,
+    EspWifiController,
 };
 
 /// A blocking HCI connector
 pub struct BleConnector<'d> {
     _device: PeripheralRef<'d, crate::hal::peripherals::BT>,
 }
 
+impl<'d> Drop for BleConnector<'d> {
+    fn drop(&mut self) {
+        crate::ble::ble_deinit();
+    }
+}
+
 impl<'d> BleConnector<'d> {
     pub fn new(
-        init: &EspWifiInitialization,
+        _init: &'d EspWifiController<'d>,
```

Comment:
Why are we taking a reference to `EspWifiController` here if it's not required?

Reply:
We need it so a user can't call `EspWifiController::deinit` whilst ble is still running. `BleConnector` borrows it for the duration of the driver's existence and you can't call `self` or `&mut self` methods whilst it's borrowed immutably.